scheduler/pkg/store/memory_status.go

```diff
@@ -105,6 +107,8 @@ func updateModelState(isLatest bool, modelVersion *ModelVersion, prevModelVersio
 }
 
 func (m *MemoryStore) FailedScheduling(modelVersion *ModelVersion, reason string, reset bool) {
+	modelVersion.mu.Lock()
+	defer modelVersion.mu.Unlock()
```

Comment:
release lock before sending to the event hub as otherwise it can cause deadlock

Reply:
caller also `scheduleToServer` is grabbing a lock on the model so it feels safe in this particular instance not to have a lock?

Reply:
Yeah, I'll remove this... it was an early change that was unrelated to the issue.